subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -748,6 +748,38 @@ static int tls_opt_ciphersuite_list_get(struct net_context *context,
 	return 0;
 }
 
+static int tls_opt_ciphersuite_used_get(struct net_context *context,
+					void *optval, socklen_t *optlen)
+{
+	const char *ciph;
+
+	if (!context || !context->tls) {
+		return -EBADF;
+	}
+
+	if (!optval || !optlen) {
+		return -EFAULT;
+	}
+
+	if (*optlen == 0) {
```

Comment:
Same thing here as with hostname, having *optlen as 1 would return empty string which is quite useless imho.

Reply:
This option will return integer, to align with TLS_CIPHERSUITE_LIST.